tpchgen-cli/src/main.rs

```diff
@@ -417,9 +417,20 @@ impl Cli {
     /// - num_parts is the total number of parts to generate
     /// - part_list is the list of parts to generate (1 based)
     fn parallel_target_part_count(&self, table: &Table) -> (i32, Vec<i32>) {
-        // parallel generation disabled if user specifies a part explicitly
+        // If a single part is specified, split it into chunks to enable parallel generation.
         if self.part != 1 || self.parts != 1 {
-            return (self.parts, vec![self.part]);
+            // Heuristic: 4 chunks per thread to maximize throughput.
```

Comment:
I don't think there is any reason to use more tasks than threads. In other words, I recommend using `num_chunks = self.num_threads`)

I tested locally and using `self.num_threads` kept all the cores busy.

Reply:
Thank you @clflushopt 

I think the integration tests are the main thing that is missing in general

It is also the thing that is missing from
- https://github.com/clflushopt/tpchgen-rs/pull/151

Reply:
I ran these integration tests and sadly they fail with this change (as you reporeted)

Reply:
FWIW I didn't carefully investigate what is different between the expected output and actual output file (e.g. is it an off by one error or something). I think it is worth exploring that -- I will do so